### DISCUSSION TITLE
How to get double click events?

### DISCUSSION THREAD

**Discussion by MauFranco** [vsg-dev/VulkanSceneGraph#1087] [Q&A]
Hi,
How to get double click events from vsg::ButtonPressEvent on Windows?
Tried using button and mask variables with no success.
Also, noticed my mouse has 5 buttons.  The button value looks the same for two of them.

**Comment by robertosfield** [comment]
The VSG's[ pointer event types](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/ui/PointerEvent.h) don't include a double click specific event type, just ButtonPressEvent and ButtonReleaseEvent.  

I have looked at the [src/vsg/platform/win32/Win32_Window.cpp]( https://github.com/vsg-dev/VulkanSceneGraph/blob/master/src/vsg/platform/win32/Win32_Window.cpp#L573) implementation there is a handling of double click but this is mapped to ButtonPressEvent. 

A quick look at the [XCB docs](https://www.x.org/releases/X11R7.5/doc/libxcb/tutorial/index.html) a double click is not event type, but suggests that users just test for button events happening within a defined time limit.  The lack of double click event under X11/XCB will be why I didn't include a specific event type for it.

We could create a new double pointer event type and have Win32_Window.cpp map to this, but we'd also need to come up with some form of emulation on the Linux and possibly macOS side in order to give users a consistent experience across platforms.

Short of that testing for multiple clicks within a "double click" period could be one way of tackling it.

**Reply by AnyOldName3** [reply]
Generally it's considered better practice to use the double click event on Windows rather than making your own as it's governed by the double click speed accessibility setting, and if you ignore it, then people with certain mobility problems won't be able to double click at all. As for non-Windows platforms, you've covered X, there doesn't appear to be any concept of a double click in Wayland, and MacOS deals with it by including a consecutive click count in its mouse events - when it's two, it's a double click - but I don't know if it's an accessibility-critical setting that controls when it resets, or just an arbitrarily-chosen value that can be safely ignored.
